[Flaking Test] UT k8s.io/kubernetes/pkg/kubelet: podcertificate

### Which jobs are flaking?

https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-kubernetes-unit/1948395184525938688
https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-kubernetes-unit-ppc64le/1947933386773368832

### Which tests are flaking?

`k8s.io/kubernetes/pkg/kubelet/podcertificate/podcertificatemanager_test.go: TestFullFlow`

### Since when has it been flaking?

Triage tool shows the earliest failures on [7/24/2025, 8:20:36 PM](https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-kubernetes-unit/1948395184525938688) as per [Link](https://storage.googleapis.com/k8s-triage/index.html?test=k8s.io%2Fkubernetes%2Fpkg%2Fkubelet%2Fpodcertificate). 

### Testgrid link

https://testgrid.k8s.io/sig-release-master-blocking#ci-kubernetes-unit

### Reason for failure (if possible)

```
 Failed
=== RUN   TestFullFlow
    podcertificatemanager_test.go:302: Error while waiting node1 podManager to know about workloadPod: context deadline exceeded
--- FAIL: TestFullFlow (15.02s)
```

### Anything else we need to know?

We were able to reproduce this flake locally using stress tool.
**_Steps to reproduce:_** 
1. Install stress tool (# go install golang.org/x/tools/cmd/stress@latest).
2. Clone the k8s repo (git clone https://github.com/kubernetes/kubernetes.git).
3. cd kubernetes
4. go test ./pkg/kubelet/podcertificate/ -c -count=1 -race
5. stress ./podcertificate.test -test.run TestFullFlow

```
[root@varsharani1 kubernetes]# uname -a
Linux varsharani1.fyre.ibm.com 5.14.0-585.el9.x86_64 #1 SMP PREEMPT_DYNAMIC Wed May 14 18:37:27 UTC 2025 x86_64 x86_64 x86_64 GNU/Linux

```
[root@varsharani1 kubernetes]# stress ./podcertificate.test -test.run TestFullFlow
5s: 16 runs so far, 0 failures, 8 active
10s: 32 runs so far, 0 failures, 8 active
15s: 54 runs so far, 0 failures, 8 active
20s: 68 runs so far, 0 failures, 8 active
25s: 78 runs so far, 0 failures, 8 active

/tmp/go-stress-20250728T020531-4126226706
I0728 02:05:42.155978 3095597 podcertificatemanager.go:274] "podcertificate.IssuingManager starting up"
I0728 02:05:42.156135 3095597 envvar.go:172] "Feature gate default state" feature="ClientsPreferCBOR" enabled=false
I0728 02:05:42.156235 3095597 envvar.go:172] "Feature gate default state" feature="InOrderInformers" enabled=true
I0728 02:05:42.156286 3095597 envvar.go:172] "Feature gate default state" feature="InformerResourceVersion" enabled=false
I0728 02:05:42.156324 3095597 envvar.go:172] "Feature gate default state" feature="WatchListClient" enabled=false
I0728 02:05:42.156356 3095597 envvar.go:172] "Feature gate default state" feature="ClientsAllowCBOR" enabled=false
```
```
[root@varsharani1 kubernetes]# cat /tmp/go-stress-20250728T020531-4126226706
I0728 02:05:42.155978 3095597 podcertificatemanager.go:274] "podcertificate.IssuingManager starting up"
I0728 02:05:42.156135 3095597 envvar.go:172] "Feature gate default state" feature="ClientsPreferCBOR" enabled=false
I0728 02:05:42.156235 3095597 envvar.go:172] "Feature gate default state" feature="InOrderInformers" enabled=true
I0728 02:05:42.156286 3095597 envvar.go:172] "Feature gate default state" feature="InformerResourceVersion" enabled=false
I0728 02:05:42.156324 3095597 envvar.go:172] "Feature gate default state" feature="WatchListClient" enabled=false
I0728 02:05:42.156356 3095597 envvar.go:172] "Feature gate default state" feature="ClientsAllowCBOR" enabled=false
I0728 02:05:42.156549 3095597 reflector.go:358] "Starting reflector" type="*v1.Pod" resyncPeriod="0s" reflector="k8s.io/client-go/informers/factory.go:160"
I0728 02:05:42.156600 3095597 reflector.go:404] "Listing and watching" type="*v1.Pod" reflector="k8s.io/client-go/informers/factory.go:160"
I0728 02:05:42.157312 3095597 reflector.go:358] "Starting reflector" type="*v1alpha1.PodCertificateRequest" resyncPeriod="24h0m0s" reflector="/root/kubernetes/test/utils/hermeticpodcertificatesigner/hermeticpodcertificatesigner.go:108"
I0728 02:05:42.157353 3095597 reflector.go:358] "Starting reflector" type="*v1alpha1.PodCertificateRequest" resyncPeriod="0s" reflector="k8s.io/client-go/informers/factory.go:160"
I0728 02:05:42.157388 3095597 reflector.go:404] "Listing and watching" type="*v1alpha1.PodCertificateRequest" reflector="/root/kubernetes/test/utils/hermeticpodcertificatesigner/hermeticpodcertificatesigner.go:108"
I0728 02:05:42.157439 3095597 reflector.go:358] "Starting reflector" type="*v1.Node" resyncPeriod="0s" reflector="k8s.io/client-go/informers/factory.go:160"
I0728 02:05:42.157429 3095597 reflector.go:404] "Listing and watching" type="*v1alpha1.PodCertificateRequest" reflector="k8s.io/client-go/informers/factory.go:160"
I0728 02:05:42.157509 3095597 reflector.go:404] "Listing and watching" type="*v1.Node" reflector="k8s.io/client-go/informers/factory.go:160"
I0728 02:05:42.157772 3095597 reflector.go:436] "Caches populated" type="*v1alpha1.PodCertificateRequest" reflector="/root/kubernetes/test/utils/hermeticpodcertificatesigner/hermeticpodcertificatesigner.go:108"
I0728 02:05:42.157804 3095597 reflector.go:436] "Caches populated" type="*v1.Pod" reflector="k8s.io/client-go/informers/factory.go:160"
I0728 02:05:42.158129 3095597 reflector.go:436] "Caches populated" type="*v1alpha1.PodCertificateRequest" reflector="k8s.io/client-go/informers/factory.go:160"
I0728 02:05:42.158301 3095597 reflector.go:436] "Caches populated" type="*v1.Node" reflector="k8s.io/client-go/informers/factory.go:160"
--- FAIL: TestFullFlow (15.01s)
    podcertificatemanager_test.go:302: Error while waiting node1 podManager to know about workloadPod: context deadline exceeded
I0728 02:05:57.159770 3095597 watch.go:218] "Stopping fake watcher"
I0728 02:05:57.159866 3095597 podcertificatemanager.go:284] "podcertificate.IssuingManager shut down"
I0728 02:05:57.159915 3095597 watch.go:218] "Stopping fake watcher"
I0728 02:05:57.159943 3095597 watch.go:218] "Stopping fake watcher"
I0728 02:05:57.160000 3095597 watch.go:218] "Stopping fake watcher"
I0728 02:05:57.160005 3095597 reflector.go:364] "Stopping reflector" type="*v1alpha1.PodCertificateRequest" resyncPeriod="24h0m0s" reflector="/root/kubernetes/test/utils/hermeticpodcertificatesigner/hermeticpodcertificatesigner.go:108"
I0728 02:05:57.160068 3095597 reflector.go:364] "Stopping reflector" type="*v1.Pod" resyncPeriod="0s" reflector="k8s.io/client-go/informers/factory.go:160"
FAIL


ERROR: exit status 1
```

### Relevant SIG(s)

/sig testing

--- a/pkg/kubelet/podcertificate/podcertificatemanager_test.go
+++ b/pkg/kubelet/podcertificate/podcertificatemanager_test.go
@@ -294,12 +294,12 @@ func TestFullFlow(t *testing.T) {
 
 	// Because our fake podManager is based on an informer, we need to poll
 	// until workloadPod is reflected in the informer.
-	err = wait.PollUntilContextTimeout(ctx, 1*time.Second, 15*time.Second, true, func(ctx context.Context) (bool, error) {
+	err = wait.PollUntilContextTimeout(ctx, 1*time.Second, 30*time.Second, true, func(ctx context.Context) (bool, error) {
 		_, ok := node1PodManager.GetPodByUID(workloadPod.ObjectMeta.UID)
 		return ok, nil
 	})
 	if err != nil {
-		t.Fatalf("Error while waiting node1 podManager to know about workloadPod: %v", err)
+		t.Fatalf("Error while waiting for node1 podManager to know about workloadPod: %v", err)
 	}
 
 	node1PodCertificateManager.TrackPod(ctx, workloadPod)
@@ -307,7 +307,7 @@ func TestFullFlow(t *testing.T) {
 	// Within a few seconds, we should see a PodCertificateRequest created for
 	// this pod.
 	var gotPCR *certsv1alpha1.PodCertificateRequest
-	err = wait.PollUntilContextTimeout(ctx, 1*time.Second, 15*time.Second, true, func(ctx context.Context) (bool, error) {
+	err = wait.PollUntilContextTimeout(ctx, 1*time.Second, 30*time.Second, true, func(ctx context.Context) (bool, error) {
 		pcrs, err := kc.CertificatesV1alpha1().PodCertificateRequests(workloadNS.ObjectMeta.Name).List(ctx, metav1.ListOptions{})
 		if err != nil {
 			return false, fmt.Errorf("while listing PodCertificateRequests: %w", err)
@@ -353,7 +353,7 @@ func TestFullFlow(t *testing.T) {
 	}
 
 	// Wait some more time for the PCR to be issued.
-	err = wait.PollUntilContextTimeout(ctx, 1*time.Second, 15*time.Second, true, func(ctx context.Context) (bool, error) {
+	err = wait.PollUntilContextTimeout(ctx, 1*time.Second, 30*time.Second, true, func(ctx context.Context) (bool, error) {
 		pcrs, err := kc.CertificatesV1alpha1().PodCertificateRequests(workloadNS.ObjectMeta.Name).List(ctx, metav1.ListOptions{})
 		if err != nil {
 			return false, fmt.Errorf("while listing PodCertificateRequests: %w", err)
@@ -388,7 +388,7 @@ func TestFullFlow(t *testing.T) {
 
 	// Now we know that the PCR was issued, so we can wait for the
 	// podcertificate manager to return some valid credentials.
-	err = wait.PollUntilContextTimeout(ctx, 1*time.Second, 15*time.Second, true, func(ctx context.Context) (bool, error) {
+	err = wait.PollUntilContextTimeout(ctx, 1*time.Second, 30*time.Second, true, func(ctx context.Context) (bool, error) {
 		_, _, err := node1PodCertificateManager.GetPodCertificateCredentialBundle(ctx, workloadPod.ObjectMeta.Namespace, workloadPod.ObjectMeta.Name, string(workloadPod.ObjectMeta.UID), "certificate", 0)
 		if err != nil {
 			return false, err
@@ -413,7 +413,7 @@ func TestFullFlow(t *testing.T) {
 
 	// Within a few seconds, we should see a new PodCertificateRequest created for
 	// this pod.
-	err = wait.PollUntilContextTimeout(ctx, 1*time.Second, 15*time.Second, true, func(ctx context.Context) (bool, error) {
+	err = wait.PollUntilContextTimeout(ctx, 1*time.Second, 30*time.Second, true, func(ctx context.Context) (bool, error) {
 		pcrs, err := kc.CertificatesV1alpha1().PodCertificateRequests(workloadNS.ObjectMeta.Name).List(ctx, metav1.ListOptions{})
 		if err != nil {
 			return false, fmt.Errorf("while listing PodCertificateRequests: %w", err)
@@ -433,7 +433,7 @@ func TestFullFlow(t *testing.T) {
 	// We will assume that the created PCR matches our expectations.
 
 	// Wait some more time for the new PCR to be issued.
-	err = wait.PollUntilContextTimeout(ctx, 1*time.Second, 15*time.Second, true, func(ctx context.Context) (bool, error) {
+	err = wait.PollUntilContextTimeout(ctx, 1*time.Second, 30*time.Second, true, func(ctx context.Context) (bool, error) {
 		pcrs, err := kc.CertificatesV1alpha1().PodCertificateRequests(workloadNS.ObjectMeta.Name).List(ctx, metav1.ListOptions{})
 		if err != nil {
 			return false, fmt.Errorf("while listing PodCertificateRequests: %w", err)
@@ -461,7 +461,7 @@ func TestFullFlow(t *testing.T) {
 
 	// Now we know that the PCR was issued, so we can wait for the
 	// podcertificate manager to start returning the new certificate
-	err = wait.PollUntilContextTimeout(ctx, 1*time.Second, 15*time.Second, true, func(ctx context.Context) (bool, error) {
+	err = wait.PollUntilContextTimeout(ctx, 1*time.Second, 30*time.Second, true, func(ctx context.Context) (bool, error) {
 		_, certChain, err := node1PodCertificateManager.GetPodCertificateCredentialBundle(ctx, workloadPod.ObjectMeta.Namespace, workloadPod.ObjectMeta.Name, string(workloadPod.ObjectMeta.UID), "certificate", 0)
 		if err != nil {
 			return false, err

